### PR TITLE
feat: adds client side query type codegen

### PR DESCRIPTION
**Pull request by gabrielcsapo**
# Summary

Adds client side typescript codgen for queries!

### CODE REVIEW

File: packages/ui/pages/index.tsx [hunk 1]
```diff
@@ -55,7 +48,7 @@ const ReportQuery = gql`
 `;
 
 interface SuggestionOverviewProps {
-  topSuggestions: NexusGenFieldTypes['TopSuggestions'][];
+  topSuggestions: IndexReportQuery['report']['topSuggestions'];
```
Comment (lbdm44) [inline packages/ui/pages/index.tsx:51]: Two Options:

1. Pull this out into a type and export it for usage via `export type TopSuggestion = IndexReportQuery['report']['topSuggestions']`
2. Use a graphql fragment and then export it for usage in multiple places

File: packages/ui/codegen.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+schema: ./graphql/generated/schema.graphql
+documents: './(pages|components)/**/*.tsx'
+generates:
+  graphql/generated.ts:
```
Comment (gabrielcsapo) [inline packages/ui/codegen.yaml:4]:
```suggestion
  graphql/generated/client.ts:
```

File: packages/ui/package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
   "scripts": {
     "dev": "yarn build:nexus && next dev",
     "build": "yarn build:nexus && next build",
-    "build:nexus": "ts-node-dev -O '{ \"module\": \"commonjs\" }' --transpile-only graphql/schema.ts && prettier --write graphql/generated/*",
+    "build:nexus": "yarn generate-graphql-types && ts-node-dev -O '{ \"module\": \"commonjs\" }' --transpile-only graphql/schema.ts && prettier --write graphql/generated/*",
+    "generate-graphql-types": "graphql-codegen",
```
Comment (lbdm44) [inline packages/ui/package.json:12]:
```suggestion
    "build:nexus": "yarn generate-graphql-types && ts-node-dev -O '{ \"module\": \"commonjs\" }' --transpile-only graphql/schema.ts",
    "build:graphql": "yarn build:nexus && yarn build:graphql-codegen && yarn prettier:generated",
    "build:graphql-codegen": "graphql-codegen",
    "prettier:graphql": "prettier --write graphql/generated/*",
```

File: packages/ui/package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
   "scripts": {
     "dev": "yarn build:nexus && next dev",
```
Comment (lbdm44) [inline packages/ui/package.json:9]: Need to change these to `yarn build:graphql && next <command>`

File: packages/ui/codegen.yaml (hunk 2 above)
Comment (gabrielcsapo) [reply packages/ui/codegen.yaml:4]: added

File: packages/ui/package.json (hunk 3 above)
Comment (gabrielcsapo) [reply packages/ui/package.json:12]: added, prettier is done at the top level.

File: packages/ui/pages/index.tsx (hunk 1 above)
Comment (gabrielcsapo) [reply packages/ui/pages/index.tsx:55]: updated, good suggestion.

**Comment by lbdm44** [comment]
@gabrielcsapo looks like type checking is failing

File: packages/ui/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export type { CardViewSuggestionsFragment } from './graphql/generated/client';
```
Comment (lbdm44) [inline packages/ui/index.ts:1]: Not something for this PR, but i dont really like that plugins are requiring the ui package, feels werid

Comment (gabrielcsapo) [reply packages/ui/index.ts:1]: Something we can make separate in the future by having a plugin abstraction for things to include instead.

File: packages/ui/package.json [hunk 6]
```diff
@@ -6,9 +6,11 @@
     "node": ">= 16"
   },
   "scripts": {
-    "dev": "yarn build:nexus && next dev",
-    "build": "yarn build:nexus && next build",
-    "build:nexus": "ts-node-dev -O '{ \"module\": \"commonjs\" }' --transpile-only graphql/schema.ts && prettier --write graphql/generated/*",
+    "dev": "yarn build:nexus && yarn build:graphql && next dev",
+    "build": "yarn build:nexus && yarn build:graphql && next build",
```
Comment (lbdm44) [inline packages/ui/package.json:10]: These are doing extra work

```suggestion
    "dev": "yarn build:graphql && next dev",
    "build": "yarn build:graphql && next build",
```

**Review by lbdm44 — APPROVED** [review]

**Pull request merged**